src/exceptions.py

```diff
@@ -13,6 +13,8 @@
 ]
 
 
+# This exception is used to signal the early termination of a reconciliation process,
+# thus not an error (N818), same for all the subclasses.
```

Comment:
Thanks for adding the note, I don't yet follow why it isn't an error? If the reconciliation process terminates early, is that not an unexpected behaviour and thus an error?

Reply:
In some cases, the reconciliation can terminates early like configuration hasn't been updated or the relation is not ready yet. The charm will go into the waiting state but I don't thing this count as an error because I think they are excepted in the early stage of the charm lifecycle. That's why I use exception instead of error here.

Reply:
Ah I see, do you think it is worth including that information in the comment? That extra information cleared it up for me

Reply:
Done